### ISSUE TITLE
Support a directory as output

### ISSUE THREAD

**Issue by vseventer** [vseventer/sharp-cli#16]
# What
Support a directory as output (`-o dir/`).

# Why
There have been multiple reports (#8, #13) where it is assumed the input files are saved in the output directory. The readme even shows this as an example (`sharp -i ./input.jpg -o ./out resize 300 200` - _`out/input.jpg` will be a [...] version of `input.jpg`_).

It makes sense for the CLI to support this ask.

# How
The following behavior is **only** true if the output directory exists. Otherwise, current behavior applies.
* `sharp -i single-file.jpg -o dir/`: single output file is `dir/single-file.jpg`.
* `sharp -i *.jpg -o dir/`: multiple output files are `dir/*.jpg`
* `sharp -i single-file-*.jpg -o dir/`: single output file is `dir/single-file-*.jpg`.

**Comment by DanielRuf** [comment]
So the current docs are wrong?

Can I provide a PR for this?

**Comment by DanielRuf** [comment]
The issue is no0t quite clear, here is the description / details about the problem:

```
sharp -i src/assets/logo.png -o src/assets/icons resize 512 512
vips__file_open_write: unable to open file "/home/druf/.../client/src/assets/icons" for writing
unix error: Is a directory
```

**Comment by DanielRuf** [comment]
The issue is not quite clear, here is the description / details about the problem:

```
sharp -i src/assets/logo.png -o src/assets/icons resize 512 512
vips__file_open_write: unable to open file "/home/druf/.../client/src/assets/icons" for writing
unix error: Is a directory
```

**Comment by vseventer** [comment]
Yeah, that is the first example under "How" in OP - in your example we'd expect the CLI to save to `src/assets/icons/logo.png`, but it attempts to save to `src.assets/icons`, which is a directory.

**Comment by winston0410** [comment]
Is this issue solved by now?  I encountered the same error @DanielRuf came across when I tried to output to a different directory.

**Comment by DanielRuf** [comment]
https://github.com/vseventer/sharp-cli/commit/c57a61537b00c03610c2aa038a37d577bdeee0ad

> [vseventer](/vseventer) closed this in ` [c57a615](/vseventer/sharp-cli/commit/c57a61537b00c03610c2aa038a37d577bdeee0ad) ` [on 15 Oct 2019](#event-2712287969)
